File: swift/NumberOfStepsToReduceANumberToZero.swift
// https://leetcode.com/problems/number-of-steps-to-reduce-a-number-to-zero/

class Solution {
    func numberOfSteps(_ num: Int) -> Int {
        var steps = 0
        var num = num
        while num > 0 {
            num = num % 2 == 0 ? num / 2 : num - 1
            steps += 1
        }
        return steps
    }
}

File: c/NumberOfStepsToReduceANumberToZero.c
// https://leetcode.com/problems/number-of-steps-to-reduce-a-number-to-zero/

int numberOfSteps(int num){
    int s = 0;
    for(int i = num; i > 0; ++s) {
        if (i%2 == 0) {
            i /= 2;
        } else {
            i--;
        }
    }
    return s;
}